### PR TITLE
test for #3840

### PR DESCRIPTION
**Pull request by jogibear9988**
#3840

### CODE REVIEW

File: Source/LinqToDB/Mapping/FluentMappingBuilder.cs [hunk 1]
```diff
@@ -22,6 +21,14 @@ public class FluentMappingBuilder
 
 		#region Init
 
+		/// <summary>
+		/// Creates new MappingSchema and fluent mapping builder for it.
+		/// </summary>
+		public FluentMappingBuilder()
```
Comment (MaceWindu) [inline Source/LinqToDB/Mapping/FluentMappingBuilder.cs:27]: Just note. FluentMappingBuilder has very unintuitive behavior when we pass mapping schema to it - it use attribute from passed schema to derive and override mapping properties from it (implemented explicitly for some attributes only like ColumnAttribute). 
We should add this information to constructors summary to make difference clear to users

Comment (igor-tkachev) [reply Source/LinqToDB/Mapping/FluentMappingBuilder.cs:27]: As we removed `mappingSchema.GetFluentMappingBulder()` method, actually, this is just a short version of 
```
var ms = new MappingSchema();
var fm = new FluentMappingBulder(ms);
```

Comment (igor-tkachev) [reply Source/LinqToDB/Mapping/FluentMappingBuilder.cs:27]: On the other hand we can use 
`new FluentMappingBulder(new())`

**Review by MaceWindu — APPROVED** [review]

**Review by sdanyliv — APPROVED** [review]

File: Source/LinqToDB/Linq/Builder/ExpressionBuilder.QueryBuilder.cs [hunk 2]
```diff
@@ -182,8 +182,9 @@ public Expression BuildExpression(IBuildContext context, Expression expression,
 									var prevCount  = ctx.SelectQuery.Select.Columns.Count;
 									var expression = ctx.BuildExpression(ma, 0, context.enforceServerSide);
 
-									if (expression.NodeType == ExpressionType.Extension && expression is DefaultValueExpression
-																						&& ma.Expression?.NodeType == ExpressionType.Parameter)
+									if (expression.NodeType == ExpressionType.Extension &&
+									    expression is DefaultValueExpression &&
+									    ma.Expression?.NodeType == ExpressionType.Parameter)
```
Comment (viceroypenguin) [inline Source/LinqToDB/Linq/Builder/ExpressionBuilder.QueryBuilder.cs:187]:
```suggestion
									if (expression is DefaultValueExpression { NodeType = ExpressionType.Extension } &&
									    ma.Expression?.NodeType == ExpressionType.Parameter)
```

File: Source/LinqToDB/Linq/Builder/TableBuilder.TableContext.cs [hunk 3]
```diff
@@ -197,7 +197,7 @@ void SetLoadWithBindings(Type objectType, ParameterExpression parentObject, List
 
 							Builder.AssociationPath.Push(Tuple.Create(new AccessorMember(ma), (IBuildContext)this, (List<LoadWithInfo[]>?)loadWith));
 
-							ex = BuildExpression(ma, 1, parentObject);
+							ex = BuildExpression(ma, 1, parentObject, true);
```
Comment (viceroypenguin) [inline Source/LinqToDB/Linq/Builder/TableBuilder.TableContext.cs:200]:
```suggestion
							ex = BuildExpression(ma, 1, parentObject, enforceServerSide: true);
```

**Review by viceroypenguin — APPROVED** [review]
PR Comment should be changed to "Fixes #3840", so that it will close the associated issue.

**Pull request merged**